firedrake/output.py

```diff
@@ -1,10 +1,109 @@
+can save a `Function` on N processes and later load it on P processes
+efficiently.
+
+Example
+-------
+In the following example we save in "example.h5" file two `Function`s,
+along with the function spaces and the mesh on which they are defined.
+Note, when saving, each function can optionally be tagged with an
+integer representing, e.g., a time step.
+
+>>> mesh = UnitSquareMesh(10, 10, name="meshA")
+>>> V = FunctionSpace(mesh, "CG", 2, name="V")
+>>> W = FunctionSpace(mesh, "CG", 1, name="W")
+>>> Z = MixedFunctionSpace((V, W), name="Z")
+>>> f = Function(V, name="fA")
+>>> g = Function(Z, name="fB")
+>>> with CheckpointFile("example.h5", 'w') as afile:
+>>>     afile.save_function(f)
+>>>     afile.save_function(g)
+
+The output file, "example.h5", would contain among others:
+
+    /
+    +-- topologies
+        +-- firedrake_mixed_function_spaces
+        |   +-- Z
+        |       | Attr: firedrake_ufl_element
+        |       +-- 0
+        |       |     Attr: firedrake_mesh: "meshA"
+        |       |     Attr: firedrake_function_space: "V"
+        |       +-- 1
+        |       |     Attr: firedrake_mesh: "meshA"
+        |       |     Attr: firedrake_function_space: "W"
+        |       +-- firedrake_functions
+        |           +-- fB
+        |               +-- 0
+        |               | Attr: firedrake_function: "fB_0"
+        |               +-- 1
+        |                 Attr: firedrake_function: "fB_1"
+        +-- meshA
+            +-- dms
+                +-- sectiondm_3_0_0_False
+                |   +-- firedrake_function_spaces
+                |   |   +-- Coordinate_function_space
+                |   |         Attr: firedrake_ufl_element
+                |   +-- vecs
+                |       +-- Coordinates
+                |             Attr: firedrake_function_space: "Coordinate_function_space"
+                |             Data: Coordinates
+                +-- sectiondm_1_0_0_False
+                |   +-- firedrake_function_spaces
+                |   |   +-- W
+                |   |         Attr: firedrake_ufl_element
+                |   +-- vecs
+                |       +-- fB_1
+                |             Attr: firedrake_function_space: "W"
+                |             Data: fB_1
+                +-- sectiondm_1_1_0_False
+                    +-- firedrake_function_spaces
+                    |   +-- V
+                    |         Attr: firedrake_ufl_element
+                    +-- vecs
+                        +-- fA
+                        |     Attr: firedrake_function_space: "V"
+                        |     Data: fA
+                        +-- fB_0
+                              Attr: firedrake_function_space: "V"
+                              Data: fB_0
+
+We can load the `Function`s that we saved in the above as in the
+following.
+
+>>> with CheckpointFile("example.h5", 'r') as afile:
+>>>     f = afile.save_function("fA", mesh_name="meshA")
```

Comment:
Here should be `f = afile.load_function("fA", mesh_name="meshA")`

Reply:
Thank you. Fixed now.